frontend/src/hooks/useActions.js

```diff
@@ -199,7 +201,13 @@ const useActions = (registerHotkeys = false) => {
       handler: () => dispatchCustomEvent('sidepanel:open', { panel: 'options' })
     },
     CONVERT_TO_DFA: {
-      // handler: () => console.log('Convert to DFA'),
+      handler: projectType === 'FSA'
+        ? () => {
+            updateProject(reorderStates(convertNFAtoDFA(reorderStates(project))))
+            console.log(reorderStates(convertNFAtoDFA(reorderStates(project))))
+            commit()
+          }
+        : undefined
```

Comment:
```suggestion
      handler: () => {
            if (projectType !== 'FSA') return
            updateProject(reorderStates(convertNFAtoDFA(reorderStates(project))))
            console.log(reorderStates(convertNFAtoDFA(reorderStates(project))))
            commit()
          }
```
Currently working on converting `useActions` to typescript and undefined functions wont be allowed.
There is also a disabled field that you can use to toggle making the action disabled